curl/inc/leatherman/curl/client.hpp

```diff
@@ -310,10 +318,11 @@ namespace leatherman { namespace curl {
          * @param file_path The file that the downloaded contents will be written to.
          * @param perms The file permissions to apply when writing to file_path.
          *              On Windows this only toggles read-only.
+         * @return Returns the HTTP response if the response status is > 400.
          */
-        void download_file(request const& req,
-                           std::string const& file_path,
-                           boost::optional<boost::filesystem::perms> perms = {});
+        boost::optional<response> download_file(request const& req,
```

Comment:
This doesn't preserve the existing binary API, which we've claimed in the past we'll try to preserve. I'd opt for adding a new function call variant that takes a `boost::optional<response>` reference.